### PR TITLE
Pass correct type to setlocale

### PR DESCRIPTION
**Pull request by theboxer**
### What does it do?
Fixes calling setlocale. `setlocale` requires a string as a second argument (https://www.php.net/manual/en/function.setlocale.php). The current code passes `null` instead of `"0"` for reading the locale value - on my machine it was causing `502 Gateway Error`.

Also set `skipEmpty` to `true` when getting the locale from system settings - as this setting is present and empty by default, so it was using the empty string value instead of the value obtained by calling `setlocale(LC_ALL, '0')`.

Tested it on php 7.4, 8.0 and 8.1 (7.4 didn't have this issue, probably because it doesn't care about argument types).

### Why is it needed?
I'm not 100% sure about the setup needed for this, but for some instances NGINX was returning 502 gateway error. Also I'd say it's better to pass correct arguments to functions :)

### How to test
Not sure :)

### Related issue(s)/PR(s)
n/a

### CODE REVIEW

**Review by Jako — APPROVED** [review]

**Review by opengeek — DISMISSED** [review]

File: core/src/Revolution/modX.php [hunk 1]
```diff
@@ -2647,12 +2647,12 @@ protected function _initCulture($options = null) {
         $this->cultureKey = $cultureKey;
         $this->setOption('cultureKey', $cultureKey);
 
-        if ($this->getOption('setlocale', $options, true)) {
```
Comment (opengeek) [inline core/src/Revolution/modX.php:2650]: Why did you remove the setlocale if statement?

Comment (theboxer) [reply core/src/Revolution/modX.php:2650]: ouch, I think that's by accident, let me fix that

**Review by opengeek — APPROVED** [review]
